app.rb

```diff
@@ -0,0 +1,90 @@
+require './classes/item'
+require './classes/book'
+require './classes/label'
+require './classes/game'
+require './classes/source'
+require './classes/author'
+
+class App
+  def initialize
+    @books = []
+    @labels = []
+    @authors = []
+  end
+
+  # list methods
+  def list_all_books
+    puts "\nListing all books:"
+    if @books.empty?
+      puts 'No books found.'
+    else
+      @books.each do |book|
+        puts "ID: #{book.id}, Author: #{book.author.first_name}, Publisher: #{book.publisher},
+        Cover state: #{book.cover_state}"
+      end
+    end
+  end
+
+  def list_all_labels
+    puts "\nListing all labels:"
+    if @labels.empty?
+      puts 'No labels found.'
+    else
+      @labels.each do |label|
+        puts "Title: #{label.title}, Color: #{label.color}"
+      end
+    end
+  end
+
+  # add methods
+  def add_book
+    puts 'Published by:'
+    publisher = gets.chomp.to_s
+    puts 'Cover state (good or bad):'
+    loop do
+      @cover_state = gets.chomp.to_s
+      break if %w[good bad].include?(@cover_state)
+      return if @cover_state == 'exit'
+
+      puts 'Invalid cover state. Please try again or type exit to return to the main menu.'
+    end
+
+    puts 'Published date (YYYY-MM-DD):'
+    loop do
+      @publish_date = gets.chomp
+      break if @publish_date.match?(/\d{4}-\d{2}-\d{2}/)
+      return if @publish_date == 'exit'
+
+      puts 'Invalid date format. Please try again.'
+    end
+
```

Comment:
It appears that there is an issue with the validation for the published date. The error message "Invalid date format. Please try again." is displayed, indicating that the input does not match the expected 'YYYY-MM-DD' format.

![image](https://github.com/George7h/Catalog-of-my-things/assets/60401323/346cc64a-71b0-4f71-bce6-f62adc916157)